coderd/workspacesearch.go

```diff
@@ -0,0 +1,80 @@
+package coderd
+
+import (
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// WorkspaceSearchQuery takes a query string and breaks it into its queryparams
+// as a set of key=value.
+func WorkspaceSearchQuery(query string) (map[string]string, error) {
+	searchParams := make(map[string]string)
+	if query == "" {
+		return searchParams, nil
+	}
+	// Because we do this in 2 passes, we want to maintain quotes on the first
+	// pass.Further splitting occurs on the second pass and quotes will be
+	// dropped.
+	elements := splitElements(query, ' ', true)
```

Comment:
I think this function can be replaced by something like this: https://go.dev/play/p/-z3GWlF-NDJ.

It appears it's duplication a lot of logic from `strings.Split`, which seems sketchy to me.

Reply:
It is going to share a lot of code with strings.Split, it is just catching quotes blocks to prevent the misunderstanding in the comment above it.

I can switch to `FieldsFunc`. The only downside to that is that it doesn't trim the quotes, but I can do that after.